src/middleware/CheckForDuplicateMiddleware.ts

```diff
@@ -0,0 +1,74 @@
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { ISignInInfo } from 'ch-node-session-handler/lib/session/model/SessionInterfaces';
+import { NextFunction, Request, Response } from 'express';
+import { UNAUTHORIZED } from 'http-status-codes';
+import { inject } from 'inversify';
+import { provide } from 'inversify-binding-decorators';
+import { BaseMiddleware } from 'inversify-express-utils';
+
+import { loggerInstance } from 'app/middleware/Logger';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { AppealsService } from 'app/modules/appeals-service/AppealsService';
+import { TOKEN_MISSING_ERROR } from 'app/utils/CommonErrors';
+import { sanitizeCompany } from 'app/utils/CompanyNumberSanitizer';
+import { getEnvOrThrow } from 'app/utils/EnvironmentUtils';
+
+const errorCustomTemplate: string = 'error-custom';
+const enquiryEmail: string = getEnvOrThrow('INQUIRY_EMAIL');
+
+const customErrorHeading = 'An Appeal has already been submitted for this penalty';
+const customErrorMessage = `If you think this is a mistake, email ${enquiryEmail}.`;
+
+@provide(CheckForDuplicateMiddleware)
+export class CheckForDuplicateMiddleware extends BaseMiddleware {
+
+    constructor(@inject(AppealsService) private readonly appealsService: AppealsService) {
+        super();
+    }
+
+    public async handler(request: Request, response: Response, next: NextFunction): Promise<void> {
+
+        if (!request.session) {
+            return next(new Error('Session is undefined'));
```

Comment:
Why not just throw an error here? - as we do in other middleware classes

Reply:
@franciscog-kainos helped me out with this one, but from my understanding it was not throwing the errors properly?

Reply:
https://companieshouse.atlassian.net/wiki/spaces/TTR/pages/1643184196/Investigation+on+LFA-1673

Reply:
middlewares in express don't handle promises correctly. If you want to notify the app that there's been an error you must call next with an error instead of throwing

Reply:
If you throw inside a promise, express will possible never all next on the middleware which will cause the program to hang

Reply:
ah i see - great that you helped solve this issue then 👍